src/javascript/javascript-import-scanner.ts

```diff
@@ -36,44 +50,109 @@ export class JavaScriptImportScanner implements JavaScriptScanner {
+        if (node.arguments.length !== 1) {
+          warnings.push(new Warning({
+            message: 'Malformed import',
+            sourceRange: document.sourceRangeForNode(node)!,
+            severity: Severity.WARNING,
+            code: 'malformed-import',
+            parsedDocument: document,
+          }));
           return;
         }
-        const source = arg.value as FileRelativeUrl;
-        if (!isPathImport(source)) {
-          // TODO(usergenic): push a warning
+        const arg = node.arguments[0];
+        if (arg.type !== 'StringLiteral') {
+          warnings.push(new Warning({
+            message:
+                'Cannot analyze dynamic imports with non-literal arguments',
+            sourceRange: document.sourceRangeForNode(node)!,
+            severity: Severity.WARNING,
+            code: 'non-literal-import',
+            parsedDocument: document,
+          }));
           return;
         }
         imports.push(new ScannedImport(
             'js-import',
-            source,
+            scanner._resolveSpecifier(
+                arg.value as string, document, node, warnings),
             document.sourceRangeForNode(node)!,
             document.sourceRangeForNode(node.callee)!,
             node,
             true));
       },
 
       enterImportDeclaration(node: babel.ImportDeclaration, _: babel.Node) {
-        const source = node.source.value as FileRelativeUrl;
-        if (!isPathImport(source)) {
-          // TODO(justinfagnani): push a warning
-          return;
-        }
         imports.push(new ScannedImport(
             'js-import',
-            source,
+            scanner._resolveSpecifier(
+                node.source.value, document, node, warnings),
             document.sourceRangeForNode(node)!,
             document.sourceRangeForNode(node.source)!,
             node,
             false));
       }
     });
-    return {features: imports, warnings: []};
+    return {features: imports, warnings};
   }
-}
 
-function isPathImport(source: string): boolean {
-  return /^(\/|\.\/|\.\.\/)/.test(source);
+  private _resolveSpecifier(
+      specifier: string, document: JavaScriptDocument, node: babel.Node,
+      warnings: any[]): FileRelativeUrl|undefined {
+    if (isPathSpecifier(specifier)) {
+      return specifier as FileRelativeUrl;
+    }
+    try {
+      new URL(specifier);
+      return specifier as FileRelativeUrl;
+    } catch (e) {
+      // not a parsable URL, try to resolve
+    }
+    if (this.moduleResolution !== 'node') {
+      warnings.push(new Warning({
+        message: 'Cannot resolve module specifier with no module resolution ' +
+            'algorithm set',
+        sourceRange: document.sourceRangeForNode(node)!,
+        severity: Severity.WARNING,
+        code: 'cant-resolve-module-specifier',
+        parsedDocument: document,
+      }));
+      return undefined;
+    }
+    const documentURL = new URL(document.url);
```

Comment:
If the JavaScriptDocument is an inline document from an HTML script tag, I think we would need to honor `baseUrl` in computing the relative URL from resolved module file of just `url`.  Does this document instance have a `baseUrl`?  If not, I think it might be in `document.parsedDocument.baseUrl`...

Reply:
Good point, I'll add a test case.

Reply:
+1, yeah, and in cases where there isn't a baseUrl, it will be the same as document.url.